### ISSUE TITLE
cannot cmd in current file whose directory starts with r

### ISSUE THREAD

**Issue by prabirshrestha** [justinmk/vim-gtfo#40]
If I have a file called `d:\r\hello.txt` and use `got` with `  let g:gtfo#terminals = { 'win' : 'cmd' }` setting. The cmd opens and closes.

I'm suspecting it has do to something with `\r` being a special character.

This seems to open correctly.

```viml
!start cmd /k "cd /d "d:/r/hello.txt""
```

`/k` in cmd says to remain in the terminal so it does not exit after the script is complete.
 `/d` in cd says to change current drive. This would avoid parsing drive letter.

**Comment by justinmk** [comment]
gtfo.vim doesn't know the parameters for the numerous terminals in existence;  they need to be provided explicitly. Try this instead:

    let g:gtfo#terminals = { 'win' : 'cmd /k' }

**Comment by prabirshrestha** [comment]
That seems to solve it. Would be good if it would be added to readme and doc.

**Comment by prabirshrestha** [comment]
Worked in vim 8 no windows. But that doesn't seem to work on neovim on windows.

Here is what I get.

![image](https://cloud.githubusercontent.com/assets/287744/22400646/a54e4516-e56f-11e6-94e8-747eb5f4c64d.png)

**Comment by justinmk** [comment]
Yep, there are some problems with shell escaping with Nvim on Windows. Please wait until 0.2. is released before using Nvim on Windows :)

**Comment by prabirshrestha** [comment]
I saw that shell escaping PRs have been merged but when I got the latest master from neovim (https://github.com/neovim/neovim/commit/58d2ce9bdbb6feab7176f451ca0248c78606aa2e) I still see the same issue. I don't see any issues/PRs tagged with `platform:windows`. Are there still issues with shell escapes?

I do have `  set shellslash` setting enabled.

**Comment by justinmk** [comment]
`set shellslash` with which 'shell' ?

**Comment by justinmk** [comment]
> Would be good if it would be added to readme and doc.

https://github.com/justinmk/vim-gtfo#settings does mention that, but I added a note to https://github.com/justinmk/vim-gtfo#platform-support also.

**Comment by prabirshrestha** [comment]
@justinmk I use `set shellslash` with cmd to convert `\` to `/` so `c:\windows\system32` looks like `c:/windows/system32`. There are lot of plugin that handles this correctly. Even windows by default supports forward slashes.

![image](https://cloud.githubusercontent.com/assets/287744/25073779/a3524990-22a2-11e7-9e15-7427053910b4.png)

**Comment by justinmk** [comment]
@prabirshrestha I thought you indicated that it's only a problem in neovim. Is it a gtfo.vim problem, present in vim and neovim? Doesn't seem to be, on my end.

>  There are lot of plugin that handles this correctly. 

Almost certainly by accident, e.g. they just happen to not send a path constructed by `fnamemodify()` to `cmd.exe`.

> Even windows by default supports forward slashes.

Mostly, yes. Try it with `del`. Hence why Nvim has to continue the silly dance.

**Comment by prabirshrestha** [comment]
In gvim/vim it works correctly on windows with `set shellslash`. The problem is only with neovim so I would consider it a bug. Let me know if you would like me to file a bug on neovim to track this.

`del a/b.txt` doesn't work because some of windows commands uses `/` as separator.

**Comment by justinmk** [comment]
@prabirshrestha If you can come up with a minimal test case a bug report at neovim repo would be most welcome.

**Comment by prabirshrestha** [comment]
@justinmk Here is the minimal vimrc.

```vim
" clone in ~/.vimplugins/vim-gtfo
" git clone https://github.com/justinmk/vim-gtfo .vimplugins/vim-gtfo
set nocompatible
syntax on
filetype plugin indent on

if has('win32') || has('win64')
    set shellslash
endif

set runtimepath+=~/.vimplugins/vim-gtfo

if has('win32') || has('win64')
  let g:gtfo#terminals = { 'win' : 'cmd /k' }
endif
```

Vim8:

```
gvim -- -u minimal-vimrc.vim
:e ~/.vimrc
gof
got
```

`gof` and `got` works in Vim8. Now try the same in neovim-qt on windows and it fails `nvim-qt -- -u minimal-vimrc.vim`. It errors with - `The syntax of the command is incorrect.` and doesn't open terminal nor the windows explorer. If I remove `set shellslash` it works in neovim.

**Comment by prabirshrestha** [comment]
@justinmk Still doesn't seem to work with latest neovim 2.2/2.3

**Comment by prabirshrestha** [comment]
@justinmk I think I finally figured out the problem.

This code doesn't work. ` set shell=$COMSPEC`

```vim
func! s:force_cmdexe() abort
  if &shell !~? "cmd" || &shellslash
    let s:shell=&shell | let s:shslash=&shellslash | let s:shcmdflag=&shellcmdflag
    set shell=$COMSPEC noshellslash shellcmdflag=/c
  endif
endf
```

Changing it to the following code seems to work. `exec 'set shell='.$COMSPEC`.

```vim
func! s:force_cmdexe() abort
  if &shell !~? "cmd" || &shellslash
    let s:shell=&shell | let s:shslash=&shellslash | let s:shcmdflag=&shellcmdflag
    exec 'set shell='.$COMSPEC
    set noshellslash shellcmdflag=/c
  endif
endf
```

Would be possible to update the plugin to use this code in the meantime while we have a proper fix for neovim?
